src/fastoad/cmd/tests/test_doe.py

```diff
@@ -0,0 +1,418 @@
+
+
+def test_write_doe_inputs_single_level(cleanup_doe_variable, cleanup, sample_variables):
+    """Test _write_doe_inputs for single-level DOE points using sample variables."""
+    destination_folder = RESULTS_FOLDER_PATH / "write_doe_inputs_single_level"
+    destination_folder.mkdir(parents=True, exist_ok=True)
+
+    # Create DOESampling instance with sample_variables and start the sampling
+    doe_config = DOESampling(
+        sampling_method="LHS",
+        variables=sample_variables,
+        destination_folder=destination_folder,
+        seed_value=12,
+    )
+    doe_config.sample_doe(sample_count=5)
+
+    # Assert
+    output_file = destination_folder / "DOE_inputs.csv"
+    assert output_file.exists(), "The output CSV file was not created."
+    written_data = pd.read_csv(output_file)
+    expected_data = pd.DataFrame(
+        {
+            "ID": [0, 1, 2, 3, 4],
+            "Var1": [
+                19.913898672550232,
+                12.526630030370269,
+                17.801429708234025,
+                14.02914992497084,
+                10.308325684759344,
+            ],
+            "Var2_pseudo": [
+                55.77441864271215,
+                48.98009939303081,
+                50.567478786760596,
+                53.56684285525269,
+                53.33749401619977,
+            ],
+            "Var3": [
+                19.913898672550232,
+                12.526630030370269,
+                17.801429708234025,
+                14.02914992497084,
+                10.308325684759344,
+            ],
+        }
+    ).set_index("ID")
+    pd.testing.assert_frame_equal(written_data.set_index("ID"), expected_data)
+
+
+def test_write_doe_inputs_multilevel(cleanup_doe_variable, cleanup, sample_variables):
+    """Test _write_doe_inputs for multi-level DOE points using sample variables."""
+    destination_folder = RESULTS_FOLDER_PATH / "write_doe_inputs_multilevel"
+    destination_folder.mkdir(parents=True, exist_ok=True)
+
+    test_level = 1
+
+    # Create DOESampling instance with sample_variables and start the sampling
+    doe_config = DOESampling(
+        sampling_method="LHS",
+        variables=sample_variables,
+        destination_folder=destination_folder,
+        seed_value=12,
+        sampling_options={"level_count": 3, "use_level": test_level},
+    )
+    doe_config.sample_doe(sample_count=5)
+
+    # Assert
+    output_file = destination_folder / f"DOE_inputs_3D_level{test_level}.csv"
+
+    assert output_file.exists()
+
+    written_data = pd.read_csv(output_file).drop("ID", axis=1)
+
+    expected_data = doe_config.doe_points_df
+
+    # We are not checking using assert_frame_equal because the column name of the output file
+    # uses pseudo, while the VariableList returned by sample_doe no
+    assert np.allclose(written_data.to_numpy(), expected_data.to_numpy(), atol=1e-4, rtol=1e-4)
+
+
+def test_DOE_from_sampled_csv(sample_csv_file):
```

Comment:
Since you are preparing for some heavy changes with linting, I suggest to changed that function name (and more specifically the upper case doe) otherwise it wont pass that rule: https://docs.astral.sh/ruff/rules/invalid-function-name/. It think there are other instances in the test file where that would cause an issue. But then again we might want to exclude test file from the inspection

Especially since in a lot of other function name you use lower case ^^'